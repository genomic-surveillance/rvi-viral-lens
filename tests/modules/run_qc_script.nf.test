nextflow_process {

    name "Test Process run_qc_script"
    script "modules/run_qc_script.nf"
    process "run_qc_script"

    test("Should run without failures") {

        when {
            params {
                // define parameters
                meta = [
                    id:"test2.9000000",
                    sample_id:"test2",
                    taxid:9000000
                ]
            }

            process {
                """
                // define inputs of the process here. Example:
                input[0]=tuple(
                    params.meta,
                    file("${projectDir}/tests/test_data/test_alignments/test2.9000000.sorted.bam"),
                    file("${projectDir}/tests/test_data/test_consensuses/test2.9000000.consensus.fa"),
                    file("${projectDir}/tests/test_data/test_references/9000000.fa"),
                    file("${projectDir}/tests/test_data/test_mpileup_out/test2.9000000.mpileup.out")
                    )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
