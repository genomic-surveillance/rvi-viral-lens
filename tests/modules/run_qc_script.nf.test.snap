{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        "test.9000000.qc.csv:md5,9ebf86f87dd785309535d2c36d292ffd",
                        "test.9000000.depth.png:md5,81848422ba9ec3bb5447faa89cbfe9ae",
                        "test.9000000,0.42,0.00,2360,32358,test.9000000.fa,test.9000000.sorted.bam,TRUE,0,5,0\r\n"
                    ]
                ]
            }
        ],
        "timestamp": "2024-03-04T17:34:48+0000"
    }
}
